moshi/src/main/java/com/squareup/moshi/BufferedSourceJsonReader.java

```diff
@@ -236,6 +237,9 @@
   }
 
   private int doPeek() throws IOException {
+    if (stackSize == 1 && buffer.indexOf(BYTE_ORDER_MARK) == 0) {
```

Comment:
This only works because the test uses a `Buffer` as its `BufferedSource`. In a streamed source this buffer will be empty unless you do `request(BYTE_ORDER_MARK.size())` first